src/bootstrap/compile.rs

```diff
@@ -106,14 +106,17 @@ pub fn std_link(build: &Build,
     t!(fs::create_dir_all(&libdir));
     add_to_sysroot(&out_dir, &libdir);
 
-    if target.contains("musl") && !target.contains("mips") {
```

Comment:
It was my understanding that we wanted to support both +crt-static and -crt-static for user code with any build of rustc. The crt_static option in config.toml only controls if rustc itself is statically or dynamically linked.

If we want to support static linking (especially with a glibc-based toolchain) on a target, we need musl-root and the startup files for that target.